src/backends/odbc/session.cpp

```diff
@@ -307,8 +307,28 @@ bool odbc_session_backend::get_last_insert_id(
             break;
           
         case prod_mssql:
-            query = "select ident_current('" + table + "')";
-            break;
+            {
+                // `ident_current()` returns seed value when it called before
+                // the actual insert, but we need (seed - increment). So use
+                // `sys.identity_columns` table to determine the last insert ID.
```

Comment:
I would just say that we can't use `ident_current()` because it doesn't distinguish between the empty table and a table with one row inserted and returns 1 in both cases, while we need 0 in the former.